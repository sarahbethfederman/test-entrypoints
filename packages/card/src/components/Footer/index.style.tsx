import styled from 'styled-components';
import { pt } from '@lendi-ui/spacing';

export const CardFooterWrapper = styled.div`
  ${pt('xxs')}
  display: flex;
  ${({ align }: { align: string }) => {
    switch (align) {
      case 'right':
        return `justify-content: flex-end;`;
      case 'center':
        return `justify-content: center;`;
      default:
        return `justify-content: flex-start;`;
    }
  }}
`;
