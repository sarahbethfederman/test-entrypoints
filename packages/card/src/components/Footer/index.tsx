import * as React from 'react';
import { LUIGlobalProps } from '@lendi-ui/utils';
import { CardFooterWrapper } from './index.style';

export type alignType = 'left' | 'center' | 'right';

export interface FooterProps extends LUIGlobalProps {
  children: React.ReactElement | React.ReactElement[] | string;
  align?: alignType;
}

const Footer = ({ children, align = 'left' }: FooterProps) => {
  return <CardFooterWrapper align={align}>{children}</CardFooterWrapper>;
};

export default Footer;
