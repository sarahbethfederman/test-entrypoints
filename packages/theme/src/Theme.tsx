import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
