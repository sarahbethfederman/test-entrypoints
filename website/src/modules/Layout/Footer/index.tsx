import * as React from 'react';
import styled from 'styled-components';
import { body } from '@lendi-ui/typography';
import { p } from '@lendi-ui/spacing';

const Wrapper = styled.div`
  ${p('xs')} ${body({
    size: 'xs',
    align: 'center',
    color: 'shade.800',
  })};
`;

export class Footer extends React.Component {
  render() {
    const CurrentYear = new Date().getFullYear();
    return (
      <Wrapper>
        Copyright &copy; <span className="curYear">{CurrentYear}</span>.
      </Wrapper>
    );
  }
}
