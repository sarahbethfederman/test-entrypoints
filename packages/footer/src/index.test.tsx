import * as React from 'react';
import { shallow } from 'enzyme';
import Theme from '@lendi-ui/theme';
import Footer from './index';

const element = shallow(
  <Theme>
    <Footer />
  </Theme>
);

describe('Footer', () => {
  it('it should mount Footer component', () => {
    expect(element.find(Footer)).toHaveLength(1);
    expect(element.find(Footer)).toMatchSnapshot();
  });
});
