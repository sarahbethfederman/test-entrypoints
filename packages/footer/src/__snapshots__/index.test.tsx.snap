// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should mount Footer component 1`] = `<WithTheme(Footer) />`;
